store/store.go

```diff
@@ -1285,6 +1285,16 @@ func findRev(needle snap.Revision, haystack []snap.Revision) bool {
 	return false
 }
 
+type ShaError struct {
```

Comment:
`HashError` would be a bit nicer here, both because it sounds nice but also because it's more general. The internals are fine, though.